app/(app)/offence/photos/index.tsx

```diff
@@ -0,0 +1,73 @@
+import { Camera, FlashMode } from 'expo-camera'
+import { Redirect } from 'expo-router'
+import { useRef, useState } from 'react'
+import { useTranslation } from 'react-i18next'
+
+import FullScreenCamera from '@/components/camera/FullScreenCamera'
+import PhotosBottomSheet from '@/components/camera/PhotosBottomSheet'
+import ScreenView from '@/components/screen-layout/ScreenView'
+import { useSnackbar } from '@/components/screen-layout/Snackbar/useSnackbar'
+import { useCameraPermission } from '@/modules/permissions/useCameraPermission'
+import { useOffenceStoreContext } from '@/state/OffenceStore/useOffenceStoreContext'
+import { useSetOffenceState } from '@/state/OffenceStore/useSetOffenceState'
+
+const MAX_PHOTOS = 4
+
+const AppRoute = () => {
+  const { t } = useTranslation()
+  const snackbar = useSnackbar()
+  useCameraPermission({ autoAsk: true })
+
+  const { setOffenceState } = useSetOffenceState()
+  const photos = useOffenceStoreContext((state) => state.photos)
+
+  const ref = useRef<Camera>(null)
+  const [flashMode, setFlashMode] = useState<FlashMode>(FlashMode.off)
+  const [loading, setLoading] = useState(false)
+
+  const takePicture = async () => {
+    setLoading(true)
+    const capturedPhoto = await ref.current?.takePictureAsync()
+
+    if (!capturedPhoto) {
+      snackbar.show(t('offenceCamera.error'), {
+        variant: 'danger',
+      })
+
+      return setLoading(false)
```

Comment:
```suggestion
      setLoading(false)
      return
```
I'd maybe go with this, as if you return something from a function it might be confusing to soemone, as to what that return type is used for

Reply:
same for the rerturn at the end of the function